rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -492,7 +492,18 @@ void FleetUpdateHandle::Implementation::bid_notice_cb(
         {make_error_str(5, "Invalid request format", e.what())}
       });
   }
+  
+  std::optional<std::string> fleet_name = std::nullopt;
+  const auto fleet_name_it = request_msg.find("fleet_name");
+  if (fleet_name_it != request_msg.end())
+  {
+    fleet_name = fleet_name_it->get<std::string>();
+  }
 
+  if (fleet_name.has_value()  && fleet_name.value() != name ){
+    return;
+  }
+  
```

Comment:
I think this block can be simplified a bit. Also add some comments and logs.
```suggestion
  //  If a fleet_name was specified in the request, only proceed if the value matches
  // the name of this fleet. 
  if (request_msg.contains("fleet_name") &&
    request_msg["fleet_name"].template get<std::string>() != name)
  {
    RCLCPP_INFO(
      node->get_logger(),
      "Ignoring BidNotice request as it is for fleet %s.,
      request_msg["fleet_name"].template get<std::string>().c_str());
    return;
  } 
```